cfme/common/physical_server_views.py

```diff
@@ -0,0 +1,197 @@
+# -*- coding: utf-8 -*-
+from lxml.html import document_fromstring
+from widgetastic.utils import (
+    Parameter,
+    Version,
+    VersionPick
+)
+from widgetastic.widget import ParametrizedView, Text, View
+from widgetastic_manageiq import (
+    BaseEntitiesView,
+    NonJSBaseEntity,
+    JSBaseEntity,
+    BaseListEntity,
+    BaseQuadIconEntity,
+    BaseTileIconEntity,
+    BootstrapTreeview,
+    BreadCrumb,
+    Button,
+    ItemsToolBarViewSelector,
+    PaginationPane,
+    SummaryTable,
+    TimelinesView,
+    BaseNonInteractiveEntitiesView
+)
+from widgetastic_patternfly import (
+    Dropdown,
+    FlashMessages,
+)
+
+from cfme.base.login import BaseLoggedInPage
+
+
+class ComputePhysicalInfrastructureServersView(BaseLoggedInPage):
+    """Common parts for server views."""
+    title = Text('.//div[@id="center_div" or @id="main-content"]//h1')
+    flash = FlashMessages(
+        './/div[@id="flash_msg_div"]/div[@id="flash_text_div" or '
+        'contains(@class, "flash_text_div")]'
+    )
+
+    @property
+    def in_compute_physical_infrastructure_servers(self):
+        def _server_page(title):
```

Comment:
title isn't used within the method, and I'd prefer to not have a method here if possible, following the style convention of the below example.

There are cases where some of the checks are version dependent, or require more complex statements than fit within the context of a return statement. These we pull out into a check before the return statement (saved and included within return), but a local method implementation would be good too.

In this case I don't think its needed, and adds unnecessary complexity.

```python
return(
    self.logged_in_as_current_user and
    self.navigation.currently_selected == ['Compute',
                                                                'Physical Infrastructure',
                                                                'Servers'])
```

Here's an example from an open PR of mine where the title check is dependant on a BZ, and I pulled the if/else block out of the return statement:
https://github.com/ManageIQ/integration_tests/pull/6354/files#diff-682d6837dd0d81e338a6e7a93b13eb94R35